### PR TITLE
vendor c/image

### PR DESCRIPTION
**Pull request by vrothberg**
To fix the registry flakes we see in #17193 making c/image more tolerant
toward unexpected EOFs from registries.

Fixes: #17193
Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```

@nalind I had to pin docker/docker as imagebuilder doesn't like the new code:
```
# github.com/openshift/imagebuilder/dockerfile/parser                                                        
vendor/github.com/openshift/imagebuilder/dockerfile/parser/parser.go:122:12: undefined: system.LCOWSupported 
vendor/github.com/openshift/imagebuilder/dockerfile/parser/parser.go:157:12: undefined: system.LCOWSupported 
```

### CODE REVIEW

**Comment by vrothberg** [comment]
@edsantiago PTAL

**Comment by vrothberg** [comment]
@containers/podman-maintainers PTAL

We can thank @mtrmac afterwards for a greener (CI) future

File: pkg/api/handlers/compat/networks.go [hunk 1]
```diff
@@ -74,6 +74,9 @@ func convertLibpodNetworktoDockerNetwork(runtime *libpod.Runtime, network *netty
 	for _, con := range cons {
 		data, err := con.Inspect(false)
 		if err != nil {
+			if errors.Is(err, define.ErrNoSuchCtr) || errors.Is(err, define.ErrCtrRemoved) {
+				continue
+			}
```
Comment (Luap99) [inline pkg/api/handlers/compat/networks.go:79]: I think this commit sneaked in, it is already merged so could you please drop it.

File: go.mod [hunk 2]
```diff
@@ -151,11 +151,12 @@ require (
...
 	github.com/skratchdot/open-golang v0.0.0-20200116055534-eef842397966 // indirect
 	github.com/stefanberger/go-pkcs11uri v0.0.0-20201008174630-78d3cae3a980 // indirect
-	github.com/sylabs/sif/v2 v2.9.0 // indirect
+	github.com/sylabs/sif/v2 v2.9.1 // indirect
 	github.com/tchap/go-patricia v2.3.0+incompatible // indirect
-	github.com/theupdateframework/go-tuf v0.5.2-0.20221207161717-9cb61d6e65f5 // indirect
+	github.com/theupdateframework/go-tuf v0.5.2 // indirect
 	github.com/titanous/rocacheck v0.0.0-20171023193734-afe73141d399 // indirect
 	github.com/vbatts/tar-split v0.11.2 // indirect
+	github.com/vbauerster/mpb/v8 v8.1.6 // indirect
```
Comment (Luap99) [inline go.mod:159]: I think it would be good to include https://github.com/containers/podman/pull/17342 here to reduce bloat.

**Comment by mtrmac** [comment]
Alternatively, #17221 contains just the one change — but depends on having a discussion about adding it to the stable branch.

**Comment by vrothberg** [comment]
> Alternatively, #17221 contains just the one change — but depends on having a discussion about adding it to the stable branch.

Closing in favor of that.

**Pull request closed without merging**